### PR TITLE
LineAttack Offsets

### PR DESCRIPTION
**Pull request by MajorCooke**
- Added offsetforward and offsetside parameters to LineAttack.
- Added LOF_ABSOFFSET and LOF_ABSPOSITION flags for LineAttack.

### CODE REVIEW

**Comment by raa-eruanna** [comment]
I have not had a chance to test this, yet, to see if it breaks anything.

But looking somewhat closely at the code, I do not see any issues. It looks like this can be merged if the tests go okay.

File: src/p_map.cpp [hunk 1]
```diff
@@ -4496,7 +4497,25 @@ AActor *P_LineAttack(AActor *t1, DAngle angle, double distance,
...
+	
+	if (flags & LAF_ABSPOSITION)
+		tempos = DVector3(offsetforward, offsetside, sz);
+	else if (flags & LAF_ABSOFFSET)
+		tempos = t1->Vec3Offset(offsetforward, offsetside, shootz);
+	else
+	{
+		double s = angle.Sin();
+		double c = angle.Cos();
+		tempos = t1->Vec3Offset(offsetforward * c + offsetside * s, offsetforward * s - offsetside * c, shootz);
```
Comment (alexey-lysiuk) [inline src/p_map.cpp]: Usage of `Vec3Offset()` here and above is wrong. This function will add `shootz` to player's Z coordinate. Although `PosAtZ()` replaces the value with its argument.
You can verify this yourself by firing regular Doom pistol with no mod(s).
These two lines needs to be adjusted to use `Vec2OffsetZ()` function.

Comment (MajorCooke) [reply src/p_map.cpp]: Done.

**Comment by MajorCooke** [comment]
...How do you dismiss a review? Or is that only for admins?

**Comment by alexey-lysiuk** [comment]
Added with minor changes in d422392.

**Pull request closed without merging**